packages/loopring_v3/contracts/aux/access/LoopringIOExchangeOwner.sol

```diff
@@ -50,9 +53,35 @@ contract LoopringIOExchangeOwner is SelectorBasedAccessManager, ERC1271, Drainab
         address[]       receivers;
     }
 
-    constructor(address _exchange)
+    // See:
+    // - https://github.com/1inch-exchange/1inchProtocol/blob/a7781cf9aa1cc2aaa5ccab0d54ecbae1327ca08f/contracts/OneSplitAudit.sol#L343
+    // - https://github.com/curvefi/curve-ren-adapter/blob/8c1fbc3fec41ebd79b06984d72ff6ace3198e62d/truffle/contracts/CurveExchangeAdapter.sol#L104
+    modifier discountCHI(address from, uint maxToBurn)
+    {
+        uint gasStart = gasleft();
+        _;
+        if (maxToBurn == 0) return;
+        uint gasSpent = 21000 + gasStart - gasleft() + 16 * msg.data.length;
+        uint fullAmountToBurn = (gasSpent + 14154) / 41947;
+        uint amountToBurn = fullAmountToBurn > maxToBurn ? maxToBurn : fullAmountToBurn;
+        if (amountToBurn > 0) {
+            chi.freeFromUpTo(from, amountToBurn);
```

Comment:
How about adding this extra logic:

``` if (from == address(0)) {
    chi.freeUpTo(amountToBurn)
} else {
...
}
```

Just in case we later move chi token into this account.